test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/DoSectionTests.java

```diff
@@ -579,82 +571,48 @@ public void testParseDoSectionAllowedWarnings() throws Exception {
         assertThat(e.getMessage(), equalTo("the warning [foo] was both allowed and expected"));
     }
 
-    public void testNodeSelectorByVersionRange() throws IOException {
+    public void testNodeSelectorByVersionRangeFails() throws IOException {
         parser = createParser(YamlXContent.yamlXContent, """
             node_selector:
                 version: 5.2.0-6.0.0
             indices.get_field_mapping:
                 index: test_index""");
 
-        DoSection doSection = DoSection.parse(parser);
-        assertNotSame(NodeSelector.ANY, doSection.getApiCallSection().getNodeSelector());
-        Node v170 = nodeWithVersion("1.7.0");
-        Node v521 = nodeWithVersion("5.2.1");
-        Node v550 = nodeWithVersion("5.5.0");
-        Node v612 = nodeWithVersion("6.1.2");
-        List<Node> nodes = new ArrayList<>();
-        nodes.add(v170);
-        nodes.add(v521);
-        nodes.add(v550);
-        nodes.add(v612);
-        doSection.getApiCallSection().getNodeSelector().select(nodes);
-        assertEquals(Arrays.asList(v521, v550), nodes);
-        ClientYamlTestExecutionContext context = mock(ClientYamlTestExecutionContext.class);
-        ClientYamlTestResponse mockResponse = mock(ClientYamlTestResponse.class);
-        when(
-            context.callApi(
-                "indices.get_field_mapping",
-                singletonMap("index", "test_index"),
-                emptyList(),
-                emptyMap(),
-                doSection.getApiCallSection().getNodeSelector()
-            )
-        ).thenReturn(mockResponse);
-        when(context.nodesVersions()).thenReturn(Set.of(randomAlphaOfLength(10)));
-        when(mockResponse.getHeaders("X-elastic-product")).thenReturn(List.of("Elasticsearch"));
-        doSection.execute(context);
-        verify(context).callApi(
-            "indices.get_field_mapping",
-            singletonMap("index", "test_index"),
-            emptyList(),
-            emptyMap(),
-            doSection.getApiCallSection().getNodeSelector()
+        var exception = expectThrows(XContentParseException.class, () -> DoSection.parse(parser));
+        assertThat(
+            exception.getMessage(),
+            endsWith("unknown version selector [5.2.0-6.0.0]. Only [current] and [non_current] are allowed.")
         );
-
-        {
-            List<Node> badNodes = new ArrayList<>();
-            badNodes.add(new Node(new HttpHost("dummy")));
-            Exception e = expectThrows(IllegalStateException.class, () -> doSection.getApiCallSection().getNodeSelector().select(badNodes));
-            assertEquals("expected [version] metadata to be set but got [host=http://dummy]", e.getMessage());
-        }
+        parser.close();
```

Comment:
Do we want to let the `AbstractClientYamlTestFragmentParserTestCase` do its check here?

Reply:
Yes, I changed it and caught a pre-existing problem. Nice!